enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotAllocation.scala

```diff
@@ -518,9 +549,11 @@ object SlotAllocation {
          _: AbstractSemiApply |
          _: AbstractSelectOrSemiApply |
          _: AbstractLetSelectOrSemiApply |
+         _: AbstractLetSemiApply |
```

Comment:
Just noticed we have very similar check here: https://github.com/neo4j/neo4j/blob/3.4/community/cypher/cypher-planner-3.4/src/main/scala/org/neo4j/cypher/internal/compiler/v3_4/planner/logical/CardinalityCostModel.scala#L156

Maybe it's time to add a trait to the plans instead of duplicating this code in at least two places?

Reply:
Ah, that would be good